src/routes/accounts.py

```
@@ -18,8 +17,193 @@
     RefreshTokenModel
 )
 from exceptions import BaseSecurityError
+from schemas.accounts import (
+    UserRegistrationRequestSchema,
+    UserRegistrationResponseSchema,
+    UserActivationRequestSchema,
+    PasswordResetRequestSchema,
+    UserLoginRequestSchema,
+    UserLoginResponseSchema,
+    TokenRefreshRequestSchema,
+    TokenRefreshResponseSchema,
+    PasswordResetCompleteRequestSchema,
+)
 from security.interfaces import JWTAuthManagerInterface
 
 router = APIRouter()
 
-# Write your code here
+
+@router.post("/register/", response_model=UserRegistrationResponseSchema, status_code=201)
+async def register(user: UserRegistrationRequestSchema, db: AsyncSession = Depends(get_db)):
+    result = await db.execute(select(UserModel).where(UserModel.email == user.email))
+    db_user_email = result.scalar_one_or_none()
+    if db_user_email:
+        raise HTTPException(status_code=409, detail=f"A user with this email {user.email} already exists.")
+
+    role = await db.execute(select(UserGroupModel).where(UserGroupModel.name == UserGroupEnum.USER))
+    db_role = role.scalar_one()
```

Role lookup is fragile: you perform `await db.execute(select(UserGroupModel).where(...))` and then `db_role = role.scalar_one()`. `scalar_one()` will raise if no row exists. Use `scalar_one_or_none()` and handle the None case (return a clear error or 500 with an explanatory message). Tests might rely on the default role being present; failing fast with a helpful message is better than an unhandled exception.